File: packages/jexlang-java/src/main/java/io/github/vinaykumar0339/eval/errors/TypeMismatchError.java
package io.github.vinaykumar0339.eval.errors;

public class TypeMismatchError extends JexLangRuntimeError {
    private final String expected;
    private final String actual;

    public TypeMismatchError(String operation, String expected, String actual) {
        super("Type mismatch in operation '" + operation + "': expected " + expected + ", but got " + actual);
        this.expected = expected;
        this.actual = actual;
    }

    public String getExpected() {
        return expected;
    }

    public String getActual() {
        return actual;
    }
}

File: packages/jexlang-swift/jexlang-swift/eval/errors/TypeMismatchError.swift
//
//  TypeMismatchError.swift
//  jexlang-swift
//
//  Created by Vinay Kumar on 03/10/25.
//

import Foundation

public class TypeMismatchError: JexLangRuntimeError {
    private let expected: String
    private let actual: String
    
    public init(operation: String, expected: String, actual: String) {
        self.expected = expected
        self.actual = actual
        super.init(message: "Type mismatch in operation '" + operation + "': expected " + expected + ", but got " + actual)
    }
    
    public func getExpected() -> String {
        return expected
    }
    
    public func getActual() -> String {
        return actual
    }
}
